frontend/src/components/FeaturedStories.jsx

```diff
@@ -11,40 +11,100 @@ const FeaturedStories = ({ user }) => {
     useEffect(() => {
         const fetchTopStories = async () => {
             try {
-                const response = await axios.get(`https://newsdata.io/api/1/latest?`, {
-                    params: {
-                        apikey: apiKey,
-                        q: `breaking`,
-                        country: 'us',
-                    }
-                });
-                setTopStories(response.data.results || []);
+                const response = await axios.get(import.meta.env.VITE_BACKEND_URL + '/api/articles');
+                setTopStories(response.data || []);
             } catch (error) {
-                console.log(apiKey);
                 console.error('Error fetching top stories:', error);
             }
         };
 
+        // Custom combination function
```

Comment:
Please add some comments on what we're trying to achieve with this combination function.

Reply:
will be in next commit